### PR TITLE
feat: add creation endpoints for the Expense

### PR DESCRIPTION
**Pull request by salet-gutierrez-jalafund**
## Trello Ticket

[ID-01](https://trello.com/c/O2qQloui)

## What did I do?

Implemented several endpoints to manage expense records. This includes:

- A `POST /api/expenses` endpoint to allow users to record a new expense, including details such as amount, description, category, date.
- A `GET /api/expenses` endpoint to retrieve all expenses recorded for the authenticated user, you can also view them by start date end date and by category, these are not mandatory as if you don't put any of them then everything will be displayed.
- A `GET /api/expenses/{id}` endpoint to get a single expense by its ID.
- A `PUT /api/expenses/{id}` endpoint to update an existing expense.
- A `DELETE /api/expenses/{id}` endpoint to delete an expense by its ID.

All interactions with the database are handled with Dapper.

## Why did I do this?

These changes were made so that the user has the ability to perform these queries such as log, get (either all or whatever the user wants), modify and delete expense records to track them. Also as developers it will help us so that we can also do these queries.

## How did I do it?

I created several new API endpoints (`POST`, `GET`, `PUT` and `DELETE`) to handle revenue records. The steps involved were:

1. Implement the database operations using Dapper to ensure efficient queries and updates to the PostgreSQL database.
2. Create appropriate response messages for success, validation errors and authorization failures.
3. Ensure that all operations are tested with tools such as Postman and Swagger to check their functionality and error handling.

## Notes:

- Some validations were added but these validations will be done in conjunction with fluentvalidation so it is not being done in this US.
- If you are testing through Swagger this would be your website link:
```
http://localhost:5149/swagger/index.html
```
But if you are testing from Postman or another platform you could use this URL:
```
http://localhost:5149/api/expenses
```
You can change it according to what you need

### CODE REVIEW

**Comment by salet-gutierrez-jalafund** [comment]
[ID-01: CRUD for Expense Management](https://trello.com/c/O2qQloui/11-id-01-crud-for-expense-management)

File: src/ExpenseTrackerGrupo2/Controllers/ExpenseController.cs [hunk 1]
```diff
@@ -1,6 +1,138 @@
+using ExpenseTrackerGrupo2.Business.Services.Interfaces;
+using ExpenseTrackerGrupo2.Business.Services.Mappers.Requests.Expenses;
+
+using Microsoft.AspNetCore.Mvc;
+
 namespace ExpenseTrackerGrupo2.Controllers;
 
-public class ExpenseController
+[ApiController]
+[Route("api/expenses")]
```
Comment (gabriela-garcia-jalafund) [inline src/ExpenseTrackerGrupo2/Controllers/ExpenseController.cs]: I suggest to change our path to api/v1/expenses

File: src/ExpenseTrackerGrupo2/Controllers/ExpenseController.cs [hunk 2]
```diff
@@ -1,6 +1,138 @@
...
 {
+    private readonly IExpenseService _expenseService;
+
+    public ExpenseController(IExpenseService expenseService)
+    {
+        _expenseService = expenseService;
+    }
+
+    [HttpPost]
+    public async Task<IActionResult> CreateExpense([FromBody] ExpenseCreateRequest request)
```
Comment (alex-paca-jalafund) [inline src/ExpenseTrackerGrupo2/Controllers/ExpenseController.cs:20]: Post method it works but display a message:
{
  "type": "https://tools.ietf.org/html/rfc9110#section-15.6.1",
  "title": "An error occurred while processing your request.",
  "status": 500
}

yes, its make the Post of the new Expense, but if it create correctly shouldnt show a error message.

**Review by alex-paca-jalafund — CHANGES_REQUESTED** [review]

File: src/ExpenseTrackerGrupo2/Controllers/ExpenseController.cs (hunk 1 above)
Comment (salet-gutierrez-jalafund) [reply src/ExpenseTrackerGrupo2/Controllers/ExpenseController.cs]: Done

File: src/ExpenseTrackerGrupo2/Controllers/ExpenseController.cs (hunk 2 above)
Comment (salet-gutierrez-jalafund) [reply src/ExpenseTrackerGrupo2/Controllers/ExpenseController.cs:20]: Done

**Review by alex-paca-jalafund — APPROVED** [review]
All operations are working except update, but that its been worked on hotfix.

**Comment by gabriela-garcia-jalafund** [comment]
Good job!

**Review by gabriela-garcia-jalafund — APPROVED** [review]

**Pull request merged**